### ISSUE TITLE
Override default context menu in tables

### ISSUE THREAD

**Issue by smellofcoffee** [ocornut/imgui#3635]
**Version/Branch of Dear ImGui:**

Version: 1.80
Branch: Table (now merged into main, but as I cloned it yesterday I do not believe there would be a difference)

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_opengl3.cpp + imgui_impl_glfw.cpp
Operating System: Windows

**My Issue/Question:**

It is currently not possible to configure a custom context menu within a table. As the comment in ImGui::TableDrawContextMenu states, "FIXME-TABLE: Ideally this should be writable by the user. Full programmatic access to that data?".

**Standalone, minimal, complete and verifiable example:**
```C++
ImGui::Begin("TestWindow");
if (ImGui::BeginTable("##table1", 3, ImGuiTableFlags_ContextMenuInBody | ImGuiTableFlags_Resizable | ImGuiTableFlags_ColumnsWidthFixed | ImGuiTableFlags_Borders | ImGuiTableFlags_RowBg))
{
    if (ImGui::BeginPopupContextItem("test_ctx_item", ImGuiPopupFlags_MouseButtonRight))
    {
        ImGui::Button("Button1");
        ImGui::Button("Button2");
        ImGui::EndPopup();
    }
    ImGui::TableSetupColumn("Col1");
    ImGui::OpenPopupOnItemClick("test_ctx_item", ImGuiPopupFlags_MouseButtonRight);
    ImGui::TableSetupColumn("Col2");
    ImGui::TableSetupColumn("Col3");
    ImGui::TableHeadersRow();

    for (int row = 0; row < 4; row++)
    {
        ImGui::TableNextRow();
        for (int column = 0; column < 3; column++)
        {
            ImGui::TableSetColumnIndex(column);
            ImGui::Text("Row %d Column %d", row, column);
            ImGui::OpenPopupOnItemClick("test_ctx_item", ImGuiPopupFlags_MouseButtonRight);
        }
    }
    ImGui::EndTable();
    ImGui::OpenPopupOnItemClick("test_ctx_item", ImGuiPopupFlags_MouseButtonRight);
}
```
As you can see from running, none of the attempts above to create a custom context menu works and are all overriden/ignored by the internal table menu. If ImGuiTableFlags_ContextMenuInBody is enabled, the table's default context menu (size to fit and so on) are shown, and if it is disabled nothing is shown. Either way, our menu is entirely ignored.

**Comment by ocornut** [comment]
Have you checked the ‘Demo->Tables->Context Menus’ section?

**Comment by smellofcoffee** [comment]
Yes and as far as I can see, one can't override the context menu entirely, the best one can do is either use a custom popup which requires adding their own handling code to check for right clicking. I was hoping for a solution that's more unified and doesn't require that like context menu for other non-table objects.
